example_test.go

```diff
@@ -0,0 +1,322 @@
+package qt_test
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"math"
+	"net"
+	"os"
+	"testing"
+
+	"github.com/go-quicktest/qt"
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+)
+
+func Example_Comment() {
+	runExampleTest(func(t testing.TB) {
+		a := 42
+		qt.Assert(t, qt.Equals(a, 42), qt.Commentf("no answer to life, the universe, and everything"))
+	})
+	// Output:
+}
+
+func Example_Equals() {
+	runExampleTest(func(t testing.TB) {
+		answer := int64(42)
+		qt.Assert(t, qt.Equals(answer, 42))
+	})
+	// Output:
+}
+
+func Example_DeepEquals() {
+	runExampleTest(func(t testing.TB) {
+		list := []int{42, 47}
+		qt.Assert(t, qt.DeepEquals(list, []int{42, 47}))
+	})
+	// Output:
+}
+
+func Example_CmpEquals() {
+	runExampleTest(func(t testing.TB) {
+		list := []int{42, 47}
+		qt.Assert(t, qt.CmpEquals(list, []int{47, 42}, cmpopts.SortSlices(func(i, j int) bool {
+			return i < j
+		})))
+	})
+	// Output:
+}
+
+type myStruct struct {
+	a int
+}
+
+func customDeepEquals[T any](got, want T) qt.Checker {
+	return qt.CmpEquals(got, want, cmp.AllowUnexported(myStruct{}))
+}
+
+func Example_CmpEquals_customfunc() {
+	runExampleTest(func(t testing.TB) {
+		got := &myStruct{
+			a: 1234,
+		}
+		qt.Assert(t, customDeepEquals(got, &myStruct{
+			a: 1234,
+		}))
+	})
+	// Output:
+}
+
+func Example_ContentEquals() {
```

Comment:
do you want the names to contain underscores? AFAIK to have an example for `Foo`, you want `ExampleFoo`, not `Example_Foo`. The end of https://pkg.go.dev/testing#hdr-Examples explains the syntax, and I think vet checks it to some degree.

Reply:
Ah, thanks! I always get this wrong and it's hard to know that it's wrong until the PR lands and one eventually sees the result inside pkg.go.dev.

Reply:
done